frontend/src/store/slices/agencies/useAgencies.ts

```diff
@@ -0,0 +1,151 @@
+import { IPagedItems } from './../../../interfaces/pagedItems';
+import { showLoading, hideLoading } from 'react-redux-loading-bar';
+import { request, success, error } from 'actions/genericActions';
+import * as actionTypes from 'constants/actionTypes';
+import * as API from 'constants/API';
+import { ENVIRONMENT } from 'constants/environment';
+import CustomAxios, { LifecycleToasts } from 'customAxios';
+import { AxiosResponse, AxiosError } from 'axios';
+import { IAddAgency, IAgency, IAgencyDetail } from 'interfaces';
+import { handleAxiosResponse } from 'utils';
+import * as pimsToasts from 'constants/toasts';
+import { useAppDispatch } from 'store/hooks';
+import { useCallback } from 'react';
+import { storeAgencies, storeAgencyDetails } from './agenciesSlice';
+
+const agencyToasts: LifecycleToasts = {
+  loadingToast: pimsToasts.agency.AGENCY_UPDATING,
+  successToast: pimsToasts.agency.AGENCY_UPDATED,
+  errorToast: pimsToasts.agency.AGENCY_ERROR,
+};
+
+/**
+ * hook that wraps calls to the agencies api.
+ */
+export const useAgencies = () => {
+  const dispatch = useAppDispatch();
+  /**
+   * fetch all of the agencies from the server based on a filter.
+   * @return the filtered, paged list of agencies.
+   */
+  const fetch = useCallback(
+    async (params: API.IPaginateParams): Promise<IPagedItems<IAgency>> => {
+      dispatch(request(actionTypes.GET_AGENCIES));
+      dispatch(showLoading());
+      return CustomAxios()
```

Comment:
Not for this PR but related to the story you're working on for Sprint 2.  Moving all API interaction into its own hook.  This would make the reducer only do reducer stuff, and provide us a single area to manage api stuff.